qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -1,136 +1,47 @@
-    in the ``optimize`` method, or, if you do not provide it, use a finite difference approximation
-    of the gradient. To adapt the size of the perturbation in the finite difference gradients,
-    set the ``perturbation`` property in the initializer.
-
-    This optimizer supports a callback function. If provided in the initializer, the optimizer
-    will call the callback in each iteration with the following information in this order:
-    current number of function values, current parameters, current function value, norm of current
-    gradient.
-
-    Examples:
-
-        A minimum example that will use finite difference gradients with a default perturbation
-        of 0.01 and a default learning rate of 0.01.
-
-        .. code-block::python
-
-            from qiskit.algorithms.optimizers import GradientDescent
-
-            def f(x):
-                return (np.linalg.norm(x) - 1) ** 2
-
-            initial_point = np.array([1, 0.5, -0.2])
-
-            optimizer = GradientDescent(maxiter=100)
-            x_opt, fx_opt, nfevs = optimizer.optimize(initial_point.size,
-                                                      f,
-                                                      initial_point=initial_point)
-
-            print(f"Found minimum {x_opt} at a value of {fx_opt} using {nfevs} evaluations.")
+from qiskit.algorithms.algorithm_result import AlgorithmResult
 
-        An example where the learning rate is an iterator and we supply the analytic gradient.
-        Note how much faster this convergences (i.e. less ``nfevs``) compared to the previous
-        example.
 
-        .. code-block::python
-
-            from qiskit.algorithms.optimizers import GradientDescent
-
-            def learning_rate():
-                power = 0.6
-                constant_coeff = 0.1
-
-                def powerlaw():
-                    n = 0
-                    while True:
-                        yield constant_coeff * (n ** power)
-                        n += 1
+from .optimizer import Optimizer, OptimizerSupportLevel, OptimizerResult, POINT
+from .steppable_optimizer import AskObject, TellObject, OptimizerState, SteppableOptimizer, CALLBACK
 
-                return powerlaw()
 
-            def f(x):
-                return (np.linalg.norm(x) - 1) ** 2
+def constant(eta=0.01):
+    """Yield a constant."""
 
-            def grad_f(x):
-                return 2 * (np.linalg.norm(x) - 1) * x / np.linalg.norm(x)
+    while True:
+        yield eta
 
-            initial_point = np.array([1, 0.5, -0.2])
 
-            optimizer = GradientDescent(maxiter=100, learning_rate=learning_rate)
-            x_opt, fx_opt, nfevs = optimizer.optimize(initial_point.size,
-                                                      f,
-                                                      gradient_function=grad_f,
-                                                      initial_point=initial_point)
+@dataclass
+class GD_OptimizerState(OptimizerState):
```

Comment:
Could we rename this e.g. to `GradientDescentState`? This would fit our naming scheme of writing things out and not using underscores a bit nicer 🙂

Reply:
Okay, thank you!